### ISSUE TITLE
Polimorfic association is incomplete

### ISSUE THREAD

**Issue by Pensarfeo** [has-friendship/has_friendship#14]
The polimorfic association we have right now does not uniquely define a relation. And can bring wrong results.
Ex
We have two tables: Men, Women
When we call find_relation we might find relationships withing Men and Women as well as Men/Men & Women/Women.

**Comment by sungwoncho** [comment]
Yes. I think a solution would be to make the `friend_type` column in the `friendship` table. Currently we only have `friend_id`. Any thoughts?

**Comment by sergeyryabov** [comment]
How about using `uuid` as primary key instead `id`?

**Comment by chevinbrown** [comment]
@Pensarfeo should a unique index fix this issue?
